src/python/pants/help/help_printer.py

```diff
@@ -96,26 +96,36 @@ def _print_goals_help(self) -> None:
             print(f"  {name.rjust(max_width)}: {description}")
         print()
 
-    def _print_options_help(self):
+    def _print_options_help(self) -> None:
         """Print a help screen.
 
         Assumes that self._help_request is an instance of OptionsHelp.
 
         Note: Ony useful if called after options have been registered.
         """
-        show_all_help = self._help_request.all_scopes
-        if show_all_help:
-            help_scopes = list(self._options.known_scope_to_info.keys())
+
+        help_request = cast(OptionsHelp, self._help_request)
+        global_options = self._options.for_global_scope()
+
+        help_scopes: Set[str]
```

Comment:
Does order matter here for stable help output? I think you can test by running `./pants help isort black docformatter pytest` multiple times (note how many scopes are there).

If it matters, use our new `pants.util.ordered_set.OrderedSet`.

Reply:
No, the order help items are printed in is enforced by code in `ScopeInfoIterator.iterate`, so it's correct for this code to not care about or try to enforce any kind of order.